### DISCUSSION TITLE
Unpredictable diff split behaviour

### DISCUSSION THREAD

**Discussion by simonmclean** [tpope/vim-fugitive#2084] [General]
A common part of my workflow before I commit is to do a quick visual scan of my changes. I do `:tab G`, move my cursor over a file, and then `dd`. But I've found the result of `dd` can be quite inconsistent in how it decides to perform the split. Usually it'll leave the status view at the top, and then open 2 buffers underneath like this (which I like)

```
| status |
----------
|  A | B |
```    

But other times it'll do 

```
| status |
----------
|   A    |
----------
|   B    |
----------
```    

And other times it'll take over the whole status view like

```
|  A  |  B  |
```    

I know I can eliminate the vertical vs horizontal inconsistency by using `dv`. But it feels like it should be more predicable, and `dv` still can have inconsistent results. 

So I'm just wondering whether this is a Vim thing, or a Fugitive thing? And whether there's some way I can mitigate it? I think what it comes down to is that there's some algorithm which decides how to lay things out, whereas I'd like to have complete control of that myself.

**Comment by tpope** [comment]
Horizontal versus vertical is determined based on the window width, [as documented](https://github.com/tpope/vim-fugitive/blob/362201bce510fce0f694093ddbac9202bedf288b/doc/fugitive.txt#L193-L195). Long term it might get phased out but I don't take changes like that lightly. If you don't like it use `dv`.

`:Git` and derivatives like `:tab G` set a special internal flag on the window that prevents "take over". If you open the buffer another way, then it won't have that. It's 100% predictable.
